pyiceberg/expressions/parser.py

```diff
@@ -84,6 +85,14 @@
 
 @column.set_parse_action
 def _(result: ParseResults) -> Reference:
+    if len(result.column) > 1:
+        deprecation_message(
+            deprecated_in="0.8.0",
+            removed_in="0.9.0",
+            help_message="Parsing expressions with table name is deprecated. Only provide field names in the row_filter.",
+        )
+    # TODO: Once this is removed, we will no longer take just the last index of parsed column result
+    # And introduce support for parsing filter expressions with nested fields.
```

Comment:
I came here to point this out indeed. Seems to fail on several levels; I don't think this was ever properly tested.

![image](https://github.com/user-attachments/assets/9d002531-08a5-4193-bd06-5af0d97ec5ef)